apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+            accountAddress,
+            allowedTokens,
+            initialYearlyUSDSalary,
+            name
+        );
+    }
+
+    /**
+     * @dev Set employee's annual salary
+     * @notice Updates also global Payroll salary sum
+     * @param employeeId Employee's identifier
+     * @param yearlyUSDSalary Employee's new salary
+     */
+    function setEmployeeSalary(uint256 employeeId, uint256 yearlyUSDSalary) public auth(OWNER_WRITE_ROLE) {
+        /* check that employee exists */
+        require(employeeIds[employees[employeeId].accountAddress] != 0);
+
+        yearlyTotalPayroll = yearlyTotalPayroll.sub(employees[employeeId].yearlyUSDSalary);
+        employees[employeeId].yearlyUSDSalary = yearlyUSDSalary;
+        yearlyTotalPayroll = yearlyTotalPayroll.add(yearlyUSDSalary);
+    }
+
+    /**
+     * @dev Remove employee from Payroll
+     * @notice Updates also global Payroll salary sum
+     * @param employeeId Employee's identifier
+     */
+    function removeEmployee(uint256 employeeId) public auth(OWNER_WRITE_ROLE) {
+        /* check that employee exists */
+        require(employeeIds[employees[employeeId].accountAddress] != 0);
+
+        yearlyTotalPayroll = yearlyTotalPayroll.sub(employees[employeeId].yearlyUSDSalary);
+        delete employeeIds[employees[employeeId].accountAddress];
+        delete employees[employeeId];
+        numEmployees--;
+    }
+
+    /**
+     * @dev Payable function to receive ETH
+     * @notice Payable function to receive ETH
+     */
+    function addFunds() public payable {
+        // convert ETH to EtherToken
+        etherToken.wrapAndCall.value(msg.value)(address(finance), "Adding Funds");
+        LogFund(
+            msg.sender,
+            address(etherToken),
+            msg.value,
+            etherToken.balanceOf(this),
+            ""
+        );
+    }
+
+    /**
+     * @dev Allows make a simple payment from this contract to Finance,
+            to avoid locked tokens in contract forever.
+            This contract should never receive tokens with a simple transfer call,
+            but in case it happens, this function allows to recover them.
+     * @notice Allows to send tokens from this contract to Finance, to avoid locked tokens in contract forever
+     */
+    function depositToFinance(address token) public auth(OWNER_WRITE_ROLE) {
+        ERC20 tokenContract = ERC20(token);
+        uint256 value = tokenContract.balanceOf(this);
+        require(value >= 0);
+
+        // make an aprrovement for the same value to Finance
+        tokenContract.approve(address(finance), value);
+        // finally deposit those tokens to Finance
+        finance.deposit(tokenContract, value, "Adding Funds");
+    }
+
+    /**
+     * @dev Implement escape hatch mechanism. Avoids locked in contract forever
+     * @notice Implement escape hatch mechanism. Avoids locked in contract forever
+     */
+    function escapeHatch() public auth(OWNER_WRITE_ROLE) {
```

Comment:
Can this be deleted now that this contract doesn't own any assets?

Reply:
Yes, with `depositToFinance` and `addFunds` we should be safe.